src/app/core/data/duplicate-search.service.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable max-classes-per-file */
+import { Observable } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { FollowLinkConfig } from '../../shared/utils/follow-link-config.model';
+import { ResponseParsingService } from './parsing.service';
+import { RemoteData } from './remote-data';
+import { GetRequest } from './request.models';
+import { RequestService } from './request.service';
+import { GenericConstructor } from '../shared/generic-constructor';
+import { HALEndpointService } from '../shared/hal-endpoint.service';
+import { SearchResponseParsingService } from './search-response-parsing.service';
+import { RemoteDataBuildService } from '../cache/builders/remote-data-build.service';
+import { RestRequest } from './rest-request.model';
+import { BaseDataService } from './base/base-data.service';
+import { FindListOptions } from './find-list-options.model';
+import { Duplicate } from '../../shared/object-list/duplicate-data/duplicate.model';
+import { PaginatedList } from './paginated-list.model';
+import { RequestParam } from '../cache/models/request-param.model';
+import { ObjectCacheService } from '../cache/object-cache.service';
+
+
+/**
+ * Service that performs all general actions that have to do with the search page
```

Comment:
This needs a better description.  It obviously is only used in the submission process to try to detect duplicates.

Optionally, it could even be moved to `/app/core/submission` and named `submission-duplicate-serch.service.ts`.... that'd be similar to how there's a `submission-cc-license-data.service.ts`, etc.

Reply:
i have moved the service to /app/core/submission and renamed it to submission-duplicate-data-service.ts